src/util-mpm-hs.c

```diff
@@ -644,6 +773,48 @@ int SCHSPreparePatterns(MpmCtx *mpm_ctx)
         PatternDatabaseFree(pd);
         SCHSFreeCompileData(cd);
         return 0;
+    } else {
+        uint32_t hash = 0;
+        hash = hashword(&pd->pattern_cnt, 1, hash);
+
+        for (uint32_t i = 0; i < pd->pattern_cnt; i++) {
+            hash = SCHSPatternHash(pd->parray[i], hash);
```

Comment:
As this depends on the rule's internal id, it will change for every rule group for many ruleset changes that are unrelated to those groups.

Say I have 1 rule:
```
http.uri; content:"blah"; sid:1;
```
Then I'll have a rule group containing that one rule, where the rule's internal id will be 0.

Now I add an unrelated rule:
```
alert icmp ... prio:1; sid:2;
http.uri; content:"blah"; sid:1;
```
Sorting will now give the icmp rule the internal id 0, and the uri rule iid 1. This give the group a new hash I think. Seems unnecessary, unless the internal id's are part of the stored state?

Reply:
relaxed the pattern hashing to only hash HS-relevant fields now